pkg/controller/intrusiondetection/intrusiondetection_controller_test.go

```diff
@@ -274,6 +274,34 @@ var _ = Describe("IntrusionDetection controller tests", func() {
 		})
 	})
 
+	Context("secret availability", func() {
+		It("should not wait on tigera-ee-installer-elasticsearch-access secret when cluster is managed", func() {
+			Expect(c.Create(ctx, &operatorv1.ManagementClusterConnection{
+				ObjectMeta: metav1.ObjectMeta{Name: "tigera-secure"},
+				Spec: operatorv1.ManagementClusterConnectionSpec{
+					ManagementClusterAddr: "127.0.0.1:12345",
+				},
+			})).ToNot(HaveOccurred())
+
+			result, err := r.Reconcile(ctx, reconcile.Request{})
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(result.Requeue).To(BeFalse())
+		})
+
+		It("should wait on tigera-ee-installer-elasticsearch-access secret when in a management cluster", func() {
+			Expect(c.Create(ctx, &operatorv1.ManagementCluster{
+				ObjectMeta: metav1.ObjectMeta{Name: "tigera-secure"},
+				Spec: operatorv1.ManagementClusterSpec{
+					Address: "127.0.0.1:12345",
+				},
+			})).ToNot(HaveOccurred())
+
+			result, err := r.Reconcile(ctx, reconcile.Request{})
+			Expect(err).ShouldNot(HaveOccurred())
```

Comment:
Shouldn't we expect an error to be returned if it's waiting for a secret?

Reply:
I'll need to rethink these tests, realized the first one doesn't fail when it is forced to check for the IDS job secret.

Reply:
I updated the tests to make sure the change is properly covered. I am now deleting the IDS job secret and expecting the management cluster to call `status.SetDegraded` in response to not finding it in `utils.ElasticSearch`.